### PR TITLE
Test switching of memory profiles

### PR DESCRIPTION
**Pull request by SergiiDmytruk**

### CODE REVIEW

File: lib/bios/menus.robot [hunk 1]
```diff
@@ -0,0 +1,77 @@
+*** Settings ***
```
Comment (macpijan) [inline lib/bios/menus.robot:1]: @SergiiDmytruk @Mixss   Is it in sync with https://github.com/Dasharo/open-source-firmware-validation/pull/25/files or you independently came up with the same file name?

Comment (SergiiDmytruk) [reply lib/bios/menus.robot:1]: Not in sync.  I deliberately used the same filename as contents is similar and this way conflicts will be obvious.

Comment (miczyg1) [reply lib/bios/menus.robot:1]: Not sure where I should start reviewing then.

**Comment by miczyg1** [comment]
Overall looks good, but I am not sure if this PR does not depend on others. WOuld be easier for me to get a hang of it if the base was not main for all PRs.

**Comment by macpijan** [comment]
BTW, it seems we fail pre-commit here: https://results.pre-commit.ci/run/github/554712842/1695990345.FXKrpGmOT5W2a-hW8fl52A

**Comment by SergiiDmytruk** [comment]
@miczyg1 @miczyg1 These are not competing changes in a sense that they serve the same purpose.  I don't know why #25 was done and how it differs from `main` (commit messages don't say much) but my changes solve a particular issue described in commit message:
> "Get Setup Menu Construction", "Get Setup Submenu Construction" and
"Enter Dasharo System Features Submenu" require that menu output wasn't
yet processed so they could grab it as the first step.
>
> This doesn't work well when the menu was already drawn.  Forcing a
redraw is problematic and causes issues with handling of the output
because several versions of the screen can be combined.
>
> This commit adds "Enter Dasharo System Features Submenu Snapshot" which
accepts a menu snapshot and updates keywords mentioned above to use it.
>
> "Get Setup Menu Construction" was also changed to call "Get Setup
Submenu Construction" as they differ in only one place.

Both changes are probably needed and will have to be combined, hence I used the same file name.

**Review by miczyg1 — APPROVED** [review]

**Comment by miczyg1** [comment]
> Thanks for this explanation @SergiiDmytruk
> 
> Based on this context, my proposal would be that we first merge your MR as it is much less invasive in changes than the one from @Mixss .
> 
> Unless you guys agree otherwise, as I am away this week, but I am sure you can come to some agreement here.
> 
> Then, we ask @Mixss to be more descriptive about the purpose of his changes and rebase on top of changes from @SergiiDmytruk

Let's do that.  
https://github.com/Dasharo/open-source-firmware-validation/pull/29 @filipleple 
https://github.com/Dasharo/open-source-firmware-validation/pull/28 https://github.com/Dasharo/open-source-firmware-validation/pull/25 https://github.com/Dasharo/open-source-firmware-validation/pull/23 @Mixss 

Please rebase these PR on main and clean up from the duplicated changes added there...

**Comment by macpijan** [comment]
I'd say that @Mixss should create a new MR first with changes to the library. After we approve that, we need to rebase the above mentioned MRs  to the current library state.

**Pull request merged**